utils/fixed_decimal/src/decimal.rs

```diff
@@ -495,6 +495,145 @@ impl FixedDecimal {
         self.check_invariants();
     }
 
+    fn n_magnitude(&self) -> i16 {
+        if self.digits.len() == 0 {
+            return self.magnitude;
+        }
+
+        self.magnitude - (self.digits.len() as i16 - 1)
+    }
+
+    fn increment_abs_by_one(&mut self) {
+        for i in (0..self.digits.len()).rev() {
+            self.digits[i] += 1;
+            if self.digits[i] < 10 {
+                return;
+            }
+            self.digits[i] = 0;
+        }
+
+        // Still there is a carry, add `1` in the beginning of the digits.
+        self.digits.insert(0, 1);
+        self.magnitude = {
+            if self.digits.len() == 1 {
+                self.magnitude
+            } else {
+                self.magnitude + 1
+            }
+        };
+
+        if self.upper_magnitude < self.magnitude {
+            self.upper_magnitude = self.magnitude;
+        }
+
+        // remove trailing zeros from `digits`
+        for i in (0..self.digits.len()).rev() {
+            if self.digits[i] == 0 {
+                self.digits.pop();
+            } else {
+                break;
+            }
+        }
+    }
+
+    /// NOTE: accept only n <= magnitude
+    fn truncate(&mut self, n: i16) {
```

Comment:
Suggestion: Make this function public and name it `truncate_right`, similar to `pad_right` and `truncate_left` which we already have.

Reply:
Done